packages/silverengine/core/src/Http/Validator.php

```
@@ -1,84 +1,92 @@
+    // Each receives ($value, $data, ...$args) and returns a string message
+    // when the rule fails, false otherwise. $data is passed so cross-field
+    // rules (e.g. 'match') can compare against sibling values.
 
-    public static function pass(): bool
+    /** @param array<string,mixed> $data */
+    private function checkRequired(mixed $value, array $data): string|false
     {
-        return empty(self::$errors);
+        unset($data);
+        return $value ? false : 'KEY is required!';
     }
```

PHPMD (2.15.0)

[warning] Avoid unused private methods such as 'checkRequired'. (undefined)

(UnusedPrivateMethod)